src/components/ReservationForm.js

```diff
@@ -0,0 +1,115 @@
+import React, { useEffect, useState } from 'react';
+import axios from 'axios';
+import { Link } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import { useCreateReservationMutation } from '../api/reservationsApi';
+import { useCurrentUserQuery } from '../api/usersData';
+
+const ReservationForm = () => {
+  const { data: currentUser } = useCurrentUserQuery();
+
+  const [createReservation] = useCreateReservationMutation();
+
+  const { user } = useSelector((store) => store.tutories);
+
+  const [reservationInfo, setReservationInfo] = useState({
+    city: '',
+    user_id: '',
+    classSubject_id: '',
+    start: '',
+    date: '',
+  });
+
+  const [classes, setClasses] = useState([]);
+
+  useEffect(() => {
+    const fetchClassesData = async () => {
+      try {
+        const response = await axios.get('http://127.0.0.1:3000/class_subjects');
```

Comment:
I kindly suggest you replace the local URLs with the URLs of the deployed API, that way the project will still work in production and not only in your local environment.